### ISSUE TITLE
gru4rec Real-Time prediction 

### ISSUE THREAD

**Issue by eveTu** [paxcema/KerasGRU4Rec#3]
Thanks for the gru4rec implementation with Keras. 
What is the best approach to real-time prediction? 

When the model is ready for real-time prediction and the sequence length are different.  How can we set the whole sequence as one input compare to set one by one item? 

Any answer will help
Thanks

**Comment by paxcema** [comment]
Hi! Thanks for the interest, and apologies for the delay in answering.

To predict you can batch several different sessions even when they have different length: you could just pad them with a padding value until they all have the same length. The network will not care about this, and will predict the next product in parallel for all sessions. If one of them is already in the padded section, you can simply ignore the output for that particular case. Another option is to reset the state and start predicting for a new session when the previous one ends, just as in the training loop.
